### PR TITLE
Fix issues with Logging module ('kolibri.lib.logging')

### LINKED ISSUE #11819: Logging module ('kolibri.lib.logging') not working as expected
## Observed behavior

There are two issues i found with **`kolibri.lib.logging`** module

1. logging module doesnt support .log function which i saw being used in some places in the codebase

https://github.com/learningequality/kolibri/blob/11233954b3ef17b56d3c23645fbd82540a6ef9c7/packages/kolibri-common/components/MeteredConnectionNotificationModal.vue#L93-L94

2.  Setlevel function of logging not working as expected (Refer Image2 attached below)

## Errors and logs

**Image 1**
image

**Image 2 (Debug statement not getting printed even after setting level to 0)**
image

## Steps to reproduce

 import the kolibri.lib.logging module and try using .log method and .setlevel methods

### LINKED ISSUE #11821: Not able to access the webpage in kolibri app_mode
## Observed behavior

On running kolibri server in [app_mode]( https://kolibri-dev.readthedocs.io/en/develop/getting_started.html#running-in-app-mode) and opening

The meter modal doesnt go away 

## Errors and logs

https://github.com/learningequality/kolibri/assets/143542192/05bce599-5696-4287-b35f-a13011455269

## Replicate

In order to replicate run the kolibri server in app mode and try opening the webpage.

## Expected behavior

To start accessing the web page after choosing the desired option.

### PR DESCRIPTION
**Pull request by iskipu**
This PR fixes both the issues mentioned in #11819  and also it fixes #11821 (the reason why the modal is not closing  is because the logging module's .log method was not working in the below mentioned lines in the code base  

https://github.com/learningequality/kolibri/blob/11233954b3ef17b56d3c23645fbd82540a6ef9c7/packages/kolibri-common/components/MeteredConnectionNotificationModal.vue#L93-L94

)

I hope this changes solves the issue.

Fixes #11819, fixes #11821 

###  Results: 
**Image (after making .log function work)**
image
**Image (after making .setLevel function work)**
image

### CODE REVIEW

**Review by rtibbles — CHANGES_REQUESTED** [review]
The logging module isn't broken. The call should be `info` not `log`. Please update the broken reference instead.

**Comment by iskipu** [comment]
> The logging module isn't broken. The call should be `info` not `log`. Please update the broken reference instead.

Hi @rtibbles since logging.js is a wrapper around npm module loglevel i thought it would be better if it directly supports .log function just like [loglevel as (mentioned here)](https://www.npmjs.com/package/loglevel#logging-methods), Plz take a look at it and if u think its not necessary i will just make changes to the reference directly.

and also any comments/suggestions regarding .setLevel function ?

**Comment by rtibbles** [comment]
It's there as an alias for debug, so if the intention is to debug then debug should be used (although this seems unintuitive because console.log is functionally more equivalent to the info method), so I don't want to expose this.

**Comment by rtibbles** [comment]
I haven't looked closely at the setLevel issue. If it's not working that's a bug in loglevel.

**Comment by iskipu** [comment]
> It's there as an alias for debug, so if the intention is to debug then debug should be used (although this seems unintuitive because console.log is functionally more equivalent to the info method), so I don't want to expose this.

Ohk  Understood, i changed the .log to .info in MeteredConnectionNotificationModal.vue like u suggested this closes #11821

**Comment by iskipu** [comment]
> I haven't looked closely at the setLevel issue. If it's not working that's a bug in loglevel.

I did check, this its not a bug in loglevel
**Headsup**: my explanation might be very bad :sweat_smile: 

the problem is that in loglevel module .debug, .info, .error ... this functions get generated dynamically based on the value of level set by setLevel (refer this [link](https://github.com/pimterry/loglevel/blob/52a80eb3bebe250a1df684bacd2b04edacddf66e/lib/loglevel.js#L95-L98))

and in our code base the following line is defined which directy uses the function (lets say x) at that point (i.e when setLevel is 3 (default value)) but since as i mentioned earlier that the functions are generated dynamically  we are stuck with x even after changing the level with the help of setLevel.
https://github.com/learningequality/kolibri/blob/171f6bb222a37fa0d338be51bf4976575e72f6f3/kolibri/core/assets/src/logging.js#L18

Inorder to fix this i proposed this changes
```python
      this[name] = param => {
          return this.logger[name].bind(console, this.messagePrefix(name))(param);
        };
```

Sorry if my explanation is bad i hope it made some sense let me know if i am doing something wrong.

**Comment by rtibbles** [comment]
Yes, you're right - it seems since this was first implemented, loglevel has implemented a plugin interface, with a specific example of adding a prefix to log statements.

I think the best thing to do would be to use this example to update the code, rather than binding to the global console!

https://npmjs.com/package/loglevel#writing-plugins

**Comment by iskipu** [comment]
> I think the best thing to do would be to use this example to update the code, rather than binding to the global console!

Hi @rtibbles great idea, made a new commit with ur suggestion.

File: kolibri/core/assets/src/logging.js [hunk 1]
```diff
@@ -5,23 +5,31 @@
...
   }
 
-  messagePrefix(type) {
-    return `[${type.toUpperCase()}: ${this.loggerName}]`;
+  setMessagePrefix() {
+    var originalFactory = this.logger.methodFactory;
+    this.logger.methodFactory = function(methodName, logLevel, loggerName) {
+      var rawMethod = originalFactory(methodName, logLevel, loggerName);
+      return function(message) {
+        rawMethod(`[${methodName}: ${loggerName}] ` + message);
```
Comment (rtibbles) [inline kolibri/core/assets/src/logging.js]: Could you add a `toUpperCase` call to the methodName to maintain the previous casing?

Comment (iskipu) [reply kolibri/core/assets/src/logging.js]: Yeah sure i overlooked it, added a new commit.

**Review by rtibbles — COMMENTED** [review]
This looks good to go, thanks @iskipu - I am going to defer merging this until after we have released 0.16.0, as the metered connection modal is not currently active in any of our products, and we are only merging critical bug fixes at this time. We should be able to merge this in a few weeks!

**Review by rtibbles — APPROVED** [review]

**Pull request merged**